TasksIT#testTransportBulkTasks can fail with an assertion error

I wasn't able to reproduce this failure locally. It looks like this test first started failing yesterday, and has failed several times since then.

---

Link to the build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=zulu11,nodes=virtual&&linux/202/

Command to reproduce: 
```
./gradlew :server:integTest \
  -Dtests.seed=6C5864EE064FF190 \
  -Dtests.class=org.elasticsearch.action.admin.cluster.node.tasks.TasksIT \
  -Dtests.method="testTransportBulkTasks" \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-QA \
  -Dtests.timezone=NST \
  -Dcompiler.java=12 \
  -Druntime.java=11
```

Relevant excerpt from the logs:
```
FAILURE 1.84s J2 | TasksIT.testTransportBulkTasks <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: expected:<1> but was:<2>
   > 	at __randomizedtesting.SeedInfo.seed([6C5864EE064FF190:80DDA65124BCAB4B]:0)
   > 	at org.elasticsearch.action.admin.cluster.node.tasks.TasksIT.testTransportBulkTasks(TasksIT.java:345)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   > 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
   > 	at java.base/java.lang.Thread.run(Thread.java:834)
```

--- a/server/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java
@@ -291,7 +291,6 @@ public class TasksIT extends ESIntegTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/37893")
     public void testTransportBulkTasks() {
         registerTaskManageListeners(BulkAction.NAME);  // main task
         registerTaskManageListeners(BulkAction.NAME + "[s]");  // shard task
@@ -299,6 +298,8 @@ public class TasksIT extends ESIntegTestCase {
         registerTaskManageListeners(BulkAction.NAME + "[s][r]");  // shard task on replica
         createIndex("test");
         ensureGreen("test"); // Make sure all shards are allocated to catch replication tasks
+        // ensures the mapping is available on all nodes so we won't retry the request (in case replicas don't have the right mapping).
+        client().admin().indices().preparePutMapping("test").setType("doc").setSource("foo", "type=keyword").get();
         client().prepareBulk().add(client().prepareIndex("test", "doc", "test_id")
             .setSource("{\"foo\": \"bar\"}", XContentType.JSON)).get();
 

